Dockerfile

```diff
@@ -1,17 +1,14 @@
-FROM golang:1.10-alpine
+FROM golang:1.10-alpine as builder
 
-RUN apk add --update make git zip
+RUN apk add --update make
 
-ENV GOPATH /usr/local/go
+COPY . /go/src/github.com/inovex/mqtt-stresser
+WORKDIR /go/src/github.com/inovex/mqtt-stresser
+RUN make linux-static
 
-RUN mkdir -p ${GOPATH}/src/github.com/inovex/ && \
-    git clone https://github.com/inovex/mqtt-stresser.git ${GOPATH}/src/github.com/inovex/mqtt-stresser/
+FROM scratch
 
-RUN cd ${GOPATH}/src/github.com/inovex/mqtt-stresser/ && \
-    make
-
-RUN cd ${GOPATH}/src/github.com/inovex/mqtt-stresser/build && \
-    tar -xzvf mqtt-stresser-linux-amd64.tar.gz && \
-    cp mqtt-stresser /bin
+COPY --from=builder /go/src/github.com/inovex/mqtt-stresser/build/mqtt-stresser.static /bin/mqtt-stresser
+COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
 
 ENTRYPOINT [ "/bin/mqtt-stresser" ]
```

Comment:
would you like to add a docker build step to the makefile and run it as part of the CI pipeline? I can give you access to Docker hub too for pushing

Reply:
please don't forget to update the image name in the README too :)

Reply:
the Dockerfile works fine 👍 

```
$ dck run --rm -it foo -broker tcp://broker.mqttdashboard.com:1883 -num-clients 100 -num-messages 10 -rampup-delay 1s -rampup-size 10 -global-timeout 180s -timeout 20s
10 worker started - waiting 1s
20 worker started - waiting 1s
30 worker started - waiting 1s
40 worker started - waiting 1s
50 worker started - waiting 1s
60 worker started - waiting 1s
70 worker started - waiting 1s
80 worker started - waiting 1s
90 worker started - waiting 1s
100 worker started
....................................................................................................
# Configuration
Concurrent Clients: 100
Messages / Client:  1000

# Results
Published Messages: 1000 (100%)
Received Messages:  1000 (100%)
Completed:          100 (100%)
Errors:             0 (0%)

# Publishing Throughput
Fastest: 26489 msg/sec
Slowest: 1762 msg/sec
Median: 11009 msg/sec

  < 11653 msg/sec  19%
  < 9180 msg/sec  11%
  < 6707 msg/sec  14%
  < 26489 msg/sec  1%
  < 21543 msg/sec  5%
  < 16598 msg/sec  7%
  < 14125 msg/sec  20%
  < 4235 msg/sec  11%
  < 28961 msg/sec  1%
  < 24016 msg/sec  2%
  < 19071 msg/sec  9%

# Receiving Througput
Fastest: 517 msg/sec
Slowest: 85 msg/sec
Median: 215 msg/sec

  < 344 msg/sec  7%
  < 301 msg/sec  5%
  < 215 msg/sec  25%
  < 171 msg/sec  21%
  < 128 msg/sec  4%
  < 561 msg/sec  1%
  < 517 msg/sec  5%
  < 474 msg/sec  3%
  < 431 msg/sec  3%
  < 388 msg/sec  7%
  < 258 msg/sec  19%
```